test/lib/jdk/test/lib/security/timestamp/TsaSigner.java

```diff
@@ -221,8 +218,7 @@ private TsaParam parseRequestParam() throws Exception {
             SignerInfo signerInfo = new SignerInfo(
                     new X500Name(issuerName),
                     signerEntry.cert.getSerialNumber(),
-                    AlgorithmId.get(
-                            AlgorithmId.getDigAlgFromSigAlg(sigAlgo)),
+                    AlgorithmId.get(SignatureUtil.extractDigestAlgFromDwithE(sigAlgo)),
```

Comment:
So, sigAlgo would never be RSASSA-PSS, EDDSA, ED25519, or ED448?

Reply:
There were no getDigestAlgInPkcs7SignerInfo in my initial version. I'll use this new method instead. I'll probably need to try if the new algorithms really work here.